utils/src/ooinstall/openshift_ansible.py

```diff
@@ -175,30 +184,20 @@ def write_proxy_settings(base_inventory):
         pass
 
 
-# pylint: disable=too-many-branches
 def write_host(host, role, inventory, schedulable=None):
     global CFG
 
     if host.preconfigured:
         return
 
     facts = ''
-    if host.ip:
-        facts += ' openshift_ip={}'.format(host.ip)
-    if host.public_ip:
-        facts += ' openshift_public_ip={}'.format(host.public_ip)
-    if host.hostname:
-        facts += ' openshift_hostname={}'.format(host.hostname)
-    if host.public_hostname:
-        facts += ' openshift_public_hostname={}'.format(host.public_hostname)
-    if host.containerized:
-        facts += ' containerized={}'.format(host.containerized)
+    for prop in HOST_VARIABLES_MAP:
```

Comment:
seems like it might be easier to push this into the Host object itself, then you could just pass in the dictionary from the config, mutate it as needed and then handle the mapping internally there. You'd have to also handle data migration within the host object itself, but seems like it would keep the logic more self-contained.